### PR TITLE
Remove Otel tracing when tracingSystem set to NONE

### PR DESCRIPTION
**Pull request by OwenCorrigan76**
Tracing is permanently enabled even when tracingSystem is set to NONE in HttpConsumer and HttpProducer. Conditional statement added so tracing is enabled only when specified in the Deployment.

### CODE REVIEW

**Review by scholzj — APPROVED** [review]
Thanks. LGTM assuming it works.

**Comment by ppatierno** [comment]
I think this deserves more investigation. I am not sure that just not setting the underneath propagator is the way to go. I mean the traces are generated but not propagated while we should understand how to not generates traces at all.
I thought that the `OpenTelemetry.noop()` returns a tracer doing nothig. We are using it in the `HttpConsumer` creation.
Even with that "noop" tracer, do we see traces? It's worth to understand how that "noop" is supposed to work than.

**Comment by ppatierno** [comment]
Can you explain exactly what's the issue you see?
I just tried running the `HttpProducer` without setting the `STRIMZI_TRACING_SYSTEM` and `OTEL_SERVICE_NAME` and I don't see any trace. It seems to me that the OTel NOOP is working as expected with no changes needed.
Even just setting `OTEL_SERVICE_NAME` doesn't initialize the tracing. The code needs `STRIMZI_TRACING_SYSTEM` to be `opentelemetry`.
I did the same with the `HttpConsumer` with same result.
I don't see any issue with the current code, unless I haven't got the issue right.
All my tries were by running clients locally together with Kafka and bridge.

**Comment by scholzj** [comment]
@ppatierno I raised the issue -> at least at that time, even without tracing enabled, it raised some warnings about being unable to connect somewhere to send the traces every few minutes.

**Comment by scholzj** [comment]
(See #118)

**Comment by OwenCorrigan76** [comment]
@ppatierno It takes around 50 - 70 messages before any tracing logs appear but they are there, even with `Opentelemetry.noop`  in use. I've been testing it locally and in a cluster with the bridge from Minikube and this appears in the logs, even when tracing is not enabled:
`Jun 12, 2023 8:04:37 AM io.opentelemetry.sdk.internal.ThrottlingLogger doLog                       │
│ SEVERE: Failed to export metrics. The request could not be executed.`


While I don't know why `noop` does not stop the tracing completely, the conditional statement I added has the desired effect of removing the openTelemetry log when tracing is not set.

**Comment by ppatierno** [comment]
I tried with a longer run and it happened after around 50 messages. I guess there should be a 60 seconds timeout for OpenTelemetry (considering I was sending one message per second).
While your solution could fix the problem, I am not still sure it's the right way to go. Did you engage with the OpenTelemetry community to ask the purpose of the Noop tracer and why it doesn't work somehow?

**Comment by OwenCorrigan76** [comment]
@ppatierno I haven't, as I thought you might know. I will engage now and see what I can find out.

**Comment by ppatierno** [comment]
I read the error better now   ... `Failed to export metrics` it's not about tracing!
Metrics and tracing in OpenTelemetry are two different things.
Metrics export is enabled by default in OTel, you need to do what we have in the bridge.

```java
System.setProperty("otel.metrics.exporter", "none"); // disable metrics
```

You can set use this line in the `HttpConsumer` and `HttpProducer` constructors.

File: java/http/java-http-producer/src/main/java/HttpProducer.java [hunk 1]
```diff
@@ -44,6 +44,7 @@ public static void main(String[] args) throws InterruptedException, URISyntaxExc
 
         HttpProducerConfig config = HttpProducerConfig.fromEnv();
         CountDownLatch messagesSentLatch = new CountDownLatch(1);
+
```
Comment (ppatierno) [inline java/http/java-http-producer/src/main/java/HttpProducer.java]: why this additional line?

File: java/http/java-http-producer/src/main/java/HttpProducer.java [hunk 2]
```diff
@@ -122,4 +124,3 @@ private void send() {
         }
     }
 }
-
```
Comment (ppatierno) [inline java/http/java-http-producer/src/main/java/HttpProducer.java:125]: why removing this line?

Comment (OwenCorrigan76) [reply java/http/java-http-producer/src/main/java/HttpProducer.java:125]: Put back one empty line. Should be correct now.

File: java/http/java-http-producer/src/main/java/HttpProducer.java (hunk 1 above)
Comment (OwenCorrigan76) [reply java/http/java-http-producer/src/main/java/HttpProducer.java]: Accidental. Now removed.

**Review by ppatierno — APPROVED** [review]
LGTM. Assuming you tried on your side as well. It worked for me.

**Review by scholzj — APPROVED** [review]
So, should this be added to the other examples as well?

**Review by scholzj — COMMENTED** [review]
So, should this be added to the other examples as well?

**Comment by OwenCorrigan76** [comment]
@ppatierno Yes, just tested again and working as expected (no metrics exported logs). @scholzj I never noticed this `failed to export metrics` log in any of the other examples. Do you mean the Vert.x example?

**Comment by scholzj** [comment]
> I never noticed this failed to export metrics log in any of the other examples. Do you mean the Vert.x example?

All the other examples use OpenTelemetry as well. So why is it not needed there as well?

**Comment by OwenCorrigan76** [comment]
@scholzj I'll deploy the Vertx and Java/Kafka Producer / Consumer and investigate whether this is an issue.

**Comment by ppatierno** [comment]
@OwenCorrigan76 @scholzj AFAICS the Vert.x examples are still using OpenTracing. No OpenTelemetry at all there, so we should not have any issue. The only issue is that we should migrate them to use OpenTelemetry soon when all Strimzi components will get rid of OpenTelemetry.
Regarding the Kafka examples, they use the OTel Kafka instrumentation library, not sure if it does help us because the OTel SDK is not directly involved. Of course, it would need testing to confirm or not.

**Comment by ppatierno** [comment]
@OwenCorrigan76 @scholzj AFAICS the Vert.x examples are still using OpenTracing. No OpenTelemetry at all there, so we should not have any issue. The only issue is that we should migrate them to use OpenTelemetry soon when all Strimzi components will get rid of OpenTelemetry.
Regarding the Kafka examples, they use the OTel Kafka instrumentation library, not sure if it does help us because the OTel SDK is not directly involved. Of course, it would need testing to confirm or not.

**Comment by ppatierno** [comment]
@OwenCorrigan76 @scholzj AFAICS the Vert.x examples are still using OpenTracing. No OpenTelemetry at all there, so we should not have any issue. The only issue is that we should migrate them to use OpenTelemetry soon when all Strimzi components will get rid of OpenTelemetry.
Regarding the Kafka examples, they use the OTel Kafka instrumentation library, not sure if it does help us because the OTel SDK is not directly involved. Of course, it would need testing to confirm or not.

**Comment by OwenCorrigan76** [comment]
@ppatierno Sorry, I forgot that Vert.x is only using OpenTracing until I went back to look at the example:
``` import io.opentracing.Span;
import io.opentracing.SpanContext;
import io.opentracing.Tracer;
import io.opentracing.propagation.Format;
import io.opentracing.propagation.TextMapAdapter;
import io.opentracing.tag.Tags;
import io.opentracing.util.GlobalTracer; 
```


I've been running the Java/ Kafka examples now for around 5 minutes with NO additional logs around metrics.

**Review by scholzj — APPROVED** [review]

**Pull request merged**